File: spark/projects/streaming-example/src/main/scala/streamingexample/Netcount.scala
package com.thedataincubator.streamingexample
import org.apache.spark.streaming.StreamingContext

object Netcount {
  def job(ssc: StreamingContext, host: String, port: Int) {
    // Create the Socket Text Stream and use it do some processing
    val wordCounts = ssc.socketTextStream(host, port)
                          .flatMap(_.split(" "))
                          .map(x => (x, 1))
                          .reduceByKey(_ + _)

    wordCounts.print()

    ssc.start()
    ssc.awaitTermination()
  }
}

File: spark/projects/py-streaming-example/netcount.py
from pyspark import SparkConf, SparkContext
from pyspark.streaming import StreamingContext

def main(port):
    conf = SparkConf().setAppName("netcount")
    sc = SparkContext(conf=conf)
    ssc = StreamingContext(sc, 1)

    word_counts = ssc.socketTextStream("localhost", port) \
                     .flatMap(lambda line: line.split(" ")) \
                     .map(lambda x: (x, 1)) \
                     .reduceByKey(lambda x, y: x + y)

    word_counts.pprint()

    ssc.start()
    ssc.awaitTermination()

if __name__ == '__main__':
    import sys
    if len(sys.argv) != 2:
        print "Must specify port to listen to."
        sys.exit(1)

    main(int(sys.argv[1]))
